src/dparse/lexer.d

```diff
@@ -1851,36 +1851,40 @@ in
 }
 do
 {
-    import std.string : lineSplitter, stripRight;
+    import std.string : chompPrefix, KeepTerminator, lineSplitter, stripRight;
 
     string leadingChars;
-    size_t i = 3;
-    size_t j;
-    bool hasOutput = false;
-    bool lastWasBlank = false;
+
+    enum LineType { none, normal, strange }
+    LineType prevLineType;
+
     switch (comment[0 .. 3])
     {
     case "///":
-        j = comment.length;
-
-        foreach (line; lineSplitter(comment))
-        {
+        foreach (line; lineSplitter!(KeepTerminator.yes)(comment))
+        {
+            immutable LineType currentLineType = line.endsWith("\n")
+                    || line.endsWith("\r")
+                    || line.endsWith("\u2028")
+                    || line.endsWith("\u2029")
+                ? LineType.normal
+                : LineType.strange;
+            if (prevLineType == LineType.strange)
+            {
+                outputRange.put(line);
+                prevLineType = currentLineType;
+                continue;
+            }
+            else
+                prevLineType = currentLineType;
             auto l = line[3 .. $];
```

Comment:
I think this should much rather be
```suggestion
            auto l = line.chompPrefix("///");
```
and the continue from the "strange line type" if should actually be removed.

When there is a strange line ending, the next line will probably be what will be most corrupted, missing the starting `///` and with this maybe even range erroring. This also means it's not really needed to have separated logic and would massively simplify this code.

A `chompPrefix` (startsWith + slice) is probably more efficient than 4 `endsWith` checks anyway